db/db.go

```diff
@@ -4,9 +4,15 @@ import (
 	"context"
 )
 
+//Storer type interface
 type Storer interface {
 	ListUsers(context.Context) ([]User, error)
+	GetUser(context.Context, int) (User, error)
+	UpdateUser(context.Context, User, int) (User, error)
```

Comment:
`UpdateUser(context.Context, User, int) (User, error)` should not return User
instead use
`UpdateUser(context.Context, User, int) (error)`